File: VB/WebSite/App_Code/DataItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Xml
Imports System.Xml.XPath

Public Class DataItem
	Private path As String

	Private _ID As Integer
	Private _Name As String
	Private _Type As String

	Public ReadOnly Property ID() As Integer
		Get
			Return Me._ID
		End Get
	End Property
	Public ReadOnly Property Name() As String
		Get
			Return Me._Name
		End Get
	End Property
	Public ReadOnly Property Type() As String
		Get
			Return Me._Type
		End Get
	End Property

	Public Sub New()
		Dim basePath As String = AppDomain.CurrentDomain.BaseDirectory
		If basePath.EndsWith("\") Then
			basePath = basePath.TrimEnd(New Char() { "\"c })
		End If
		Me.path = String.Format("{0}/XMLFile.xml", basePath)
	End Sub

	Public Sub New(ByVal id As Integer, ByVal name As String, ByVal type As String)
		Me._ID = id
		Me._Name = name
		Me._Type = type
	End Sub

	Public Function SelectAllItems() As DataTable
		Dim dataTable As New DataTable("DataTable")

		dataTable.Columns.Add("ID", GetType(Integer))
		dataTable.Columns.Add("Name", GetType(String))
		dataTable.Columns.Add("Type", GetType(String))

		dataTable.PrimaryKey = New DataColumn() { dataTable.Columns("ID") }

		Dim document As New XmlDocument()
		document.Load(Me.path)

		Dim navigator As XPathNavigator = document.CreateNavigator()
		Dim nodes As XPathNodeIterator = navigator.Select("descendant::Package")

		Do While nodes.MoveNext()
			Dim id As Integer = nodes.Current.SelectSingleNode("ID").ValueAsInt
			Dim name As String = nodes.Current.SelectSingleNode("Name").Value
			Dim type As String = nodes.Current.SelectSingleNode("Type").Value
			dataTable.Rows.Add(New Object() { id, name, type })
		Loop

		Return dataTable
	End Function
End Class

File: CS/WebSite/App_Code/DataItem.cs
using System;
using System.Data;
using System.Xml;
using System.Xml.XPath;

public class DataItem {
    private string path;

    private int _ID;
    private string _Name;
    private string _Type;

    public int ID { get { return this._ID; } }
    public string Name { get { return this._Name; } }
    public string Type { get { return this._Type; } }

    public DataItem() {
        string basePath = AppDomain.CurrentDomain.BaseDirectory;
        if (basePath.EndsWith("\\")) 
            basePath = basePath.TrimEnd(new char[] { '\\' });
        this.path = string.Format("{0}/XMLFile.xml", basePath);
    }

    public DataItem(int id, string name, string type) {
        this._ID = id;
        this._Name = name;
        this._Type = type;
    }

    public DataTable SelectAllItems() {
        DataTable dataTable = new DataTable("DataTable");

        dataTable.Columns.Add("ID", typeof(int));
        dataTable.Columns.Add("Name", typeof(string));
        dataTable.Columns.Add("Type", typeof(string));

        dataTable.PrimaryKey = new DataColumn[] { dataTable.Columns["ID"] };

        XmlDocument document = new XmlDocument();
        document.Load(this.path);

        XPathNavigator navigator = document.CreateNavigator();
        XPathNodeIterator nodes = navigator.Select("descendant::Package");

        while (nodes.MoveNext()) {
            int id = nodes.Current.SelectSingleNode("ID").ValueAsInt;
            string name = nodes.Current.SelectSingleNode("Name").Value;
            string type = nodes.Current.SelectSingleNode("Type").Value;
            dataTable.Rows.Add(new object[] { id, name, type });
        }

        return dataTable;
    }
}